### PR TITLE
Bilibili Comics Add search from Api instead of scrapping

### PR DESCRIPTION
**Pull request by FourTOne5**

### CODE REVIEW

File: src/en/bilibilicomics/src/eu/kanade/tachiyomi/extension/en/bilibilicomics/BilibiliComics.kt [hunk 1]
```diff
@@ -124,42 +116,34 @@ class BilibiliComics : HttpSource() {
...
+        val comicList = jsonResponse["data"]["list"].array
+            .map(::searchMangaFromObject)
 
         return MangasPage(comicList, hasNextPage = false)
     }
 
     private fun searchMangaFromObject(obj: JsonElement): SManga = SManga.create().apply {
-        title = obj["title"].string
+        //Results to "Me and <em class=\"keyword\">My</em> Zoo" (to highlight search query) I have 0 knowledge what to do here + on phone.
+        title = obj["title"].string.text() //.text() should fix it.
```
Comment (jopejoe1) [inline src/en/bilibilicomics/src/eu/kanade/tachiyomi/extension/en/bilibilicomics/BilibiliComics.kt:144]: Should be parsed to Jsoup before using `.text()`

**Comment by alessandrojean** [comment]
As I commented on my fork, I will test your changes and check if it's working. I will submit a review here.

File: src/en/bilibilicomics/src/eu/kanade/tachiyomi/extension/en/bilibilicomics/BilibiliComics.kt [hunk 2]
```diff
@@ -96,26 +96,18 @@ class BilibiliComics : HttpSource() {
     }
 
     override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
```
Comment (alessandrojean) [inline src/en/bilibilicomics/src/eu/kanade/tachiyomi/extension/en/bilibilicomics/BilibiliComics.kt:98]: Can remove this function now.

**Review by alessandrojean — CHANGES_REQUESTED** [review]

**Comment by alessandrojean** [comment]
You can check the [commit on my fork](https://github.com/alessandrojean/tachiyomi-extensions/commit/180e333a5ed9b334f1c5fef6d1e32496a2612cfe) as a reference too.

**Comment by FourTOne5** [comment]
This should do it then.

**Review by alessandrojean — APPROVED** [review]

**Pull request merged**